### PR TITLE
Refactor Kueue module

### PR DESCRIPTION
**Pull request by Ygnas**
# Issue link

Closes: [RHOAIENG-13331](https://issues.redhat.com/browse/RHOAIENG-13331)

- [x] #695

# What changes have been made

Refactored the Kueue module to fit the design doc

# Verification steps

E2E and Unit tests should be sufficient

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Review by Bobbins228 — COMMENTED** [review]
/lgtm happy to merge when preceding PR is :+1:

File: src/codeflare_sdk/ray/cluster/generate_yaml.py [hunk 1]
```diff
@@ -25,11 +25,11 @@
 import uuid
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk/ray/cluster/generate_yaml.py:25]: You can remove the import above `from typing import Optional`

**Review by ChristianZaccaria — COMMENTED** [review]
/lgtm just a non-blocking nitpick. - Good for merge

**Review by ChristianZaccaria — APPROVED** [review]

**Pull request merged**